TodoApp/test/TodoApp.Api.Tests/Services/UrlGeneratorTests.test.cs

```diff
@@ -35,5 +29,15 @@ public void GetItemUrl_UrlReceived()
 
             Assert.That(receivedUrl, Is.EqualTo(requestedUrl));
         }
+
+        private static bool ContainsCorrectId(object first)
+        {
+            new RouteValueDictionary(first).TryGetValue("id", out var firstId);
+
+            if (firstId != null && firstId.Equals(Guid.Parse("5f6a2723-040a-4398-8b63-9d55153378ba")))
```

Comment:
Since you use this string with a GUID (or even better: the whole GUID) multiple times, it would be great to create a constant (or a static read-only field) for it...

Reply:
The `TryGetValue` returns a boolean that determines whether some `value` was found under given `key`. You can simply call `firstId* == Guid.Parse(…)` and omit the `null-check` - this combined could make your method more readable.

* - `firstId` is not a nice name - `id` or `itemId` would be a bit more descriptive...